execution/src/types.rs

```diff
@@ -34,3 +35,35 @@ impl fmt::Debug for CallOpts {
             .finish()
     }
 }
+
+/// The [`ProofVerificationInput`] struct is used
+/// to encapsulate the input parameters required for verifying
+/// a Merkle proof in the context of Ethereum storage.
+/// It contains the proof path, the root hash, the storage key hash, and the corresponding value.
+/// # Fields
+/// `proof` – An array of RLP-serialized MerkleTree nodes.
+/// This array starts with the `storageHash` node and follows the path of the SHA3 hash of the key.
+/// This represents the proof path from the storage root to the desired storage value.
+///
+/// `root` – A 32-byte vector representing the SHA3 hash of the StorageRoot.
+/// All storage data will be delivered to a Merkle proof that starts with this root hash.
+/// This serves as the anchor for verifying the proof.
+///
+/// `path` – A vector of bytes representing the requested storage key hash.
+/// This is the specific key within the storage trie for which the proof is being verified.
+///
+/// `value` – A vector of bytes representing the storage value.
+/// This is the value stored at the location identified by the key hash in the path field.
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct ProofVerificationInput {
```

Comment:
```suggestion
pub struct AccountProof {
```
Is this not a more suitable name?

Reply:
I see its also used as a `StorageProof`. Perhaps just `Proof`?

Reply:
`Proof` would be the best, but it is already in use inside helios, so i thought this would not be a good idea.
Since this struct is used as the input for the verification function only, I could not thought of a better name for this.